### DISCUSSION TITLE
light version of powershell

### DISCUSSION THREAD

**Discussion by tigerinus** [PowerShell/PowerShell#17993] [Enhancements]
The project I am working on involves setup and migration, currently implemented in BASH. 

Due to the high maintenance cost of BASH scripts (debug, test, etc.) we are looking for alternative shell scripting languages.

It has to be light weight single binary (under 20mb - the less the better) that can be distributed along with our .tar.gz, and it should not involve installing any external dependencies. That rules out languages like Python, Perl, NodeJS and Google's ZX. (We could use Python or Perl, but that assumes the target system comes with one of the interpreters)

Due to that the setup and migration is mostly about file operation and file downloading, I wonder would it be a great idea to have a lightweight version of PowerShell, a single binary with just super basic cmdlets, just for this purpose?

**Comment by kasini3000** [comment]
Please allow me to give you a "community fans" level experience.

Powershell can fully meet your needs, and does not require any special lightweight version.
The Linux version of powershell is basically green software.

1 First, you need to install the libicu, openssl library of the distribution.

2Download the general linux package, decompress it, and set the soft link of pwsh. Then give pwsh execute permission.
eg: 
https://github.com/PowerShell/PowerShell/releases/download/v7.2.6/powershell-7.2.6-linux-x64.tar.gz

3 I made a script for batch operations, you can refer to:
https://gitee.com/chuanjiao10/kasini3000/blob/master/zkj_install_powershell_offline_win2linux_keyfile.ps1

it is copy powershell files from "master" to "node", IT is "one" to "more" .If you want to install powershell in batches by copying files.

the other way:  
1 https://github.com/PowerShell/PowerShell/blob/master/tools/install-powershell.sh
2 search:   One-click one key install powershell linux 一键 安装 debian ubuntu centos alpine


Finally, it should be noted that:
At present, there are only packages for x86, arm cpu architecture, and other types of linux, which need to be compiled by themselves.

**Reply by tigerinus** [reply]
> 1 First, you need to install the libicu, openssl library of the distribution.

This step disqualifies powershell to meet my requirement. It means as part of our setup and migration, we will have to be responsible to install these two across all distributions.

I'm looking for a single binary without extra dependencies.

**Comment by kasini3000** [comment]
when you use dnf ,yum,or apt,dependencies will be installed also.

Can you talk about your use case?
powershell can run in a docker container,
I also have a master-slave architecture, powershell automation solution

**Reply by tigerinus** [reply]
The use case is already described in the issue. We need a minimum scripting to replace bash. It has to be as light weight as possible so it can be packaged with our installer. It should not require any external dependencies to be installed before running the installer.

It's OK. We are now looking into Lua, which is small enough to meet our requirements.

Thanks.
